pkg/proxy/iptables/proxier.go

```diff
@@ -577,21 +582,35 @@ func (proxier *Proxier) buildEndpointInfoList(endPoints []hostPortInfo, endpoint
 	return filteredEndpoints
 }
 
-// OnEndpointsUpdate takes in a slice of updated endpoints.
-func (proxier *Proxier) OnEndpointsUpdate(allEndpoints []api.Endpoints) {
-	start := time.Now()
-	defer func() {
-		glog.V(4).Infof("OnEndpointsUpdate took %v for %d endpoints", time.Since(start), len(allEndpoints))
-	}()
+// Map of namespaced names (eg, service namespace/name) of a service to either
+// an empty set, or a set container exactly the same namespaced service name as
+// a string when the service has local endpoints.  We need this later because
+// healthcheck.UpdateEndpoints() is a silly API.
+type healthCheckMap map[types.NamespacedName]sets.String
 
-	proxier.mu.Lock()
-	defer proxier.mu.Unlock()
-	proxier.haveReceivedEndpointsUpdate = true
+func addHealthCheckUpdate(hcMap healthCheckMap, name types.NamespacedName, hostPorts []hostPortInfo) {
```

Comment:
I know it's not required by language and it makes things  wee bit more syntactically heavy, but I prefer passing maps that we expect the callee to modify by pointer, so it's clearer at the callsite